src/hotspot/os/windows/os_windows.cpp

```diff
@@ -4999,9 +4999,10 @@ void os::pause() {
 
 Thread* os::ThreadCrashProtection::_protected_thread = NULL;
 os::ThreadCrashProtection* os::ThreadCrashProtection::_crash_protection = NULL;
-volatile intptr_t os::ThreadCrashProtection::_crash_mux = 0;
 
 os::ThreadCrashProtection::ThreadCrashProtection() {
+  assert(Thread::current()->is_JfrSampler_thread(), "should be JFRSampler");
+  _protected_thread = Thread::current();
```

Comment:
Perhaps this:
  `_protected_thread = Thread::current();`
  `assert(_protected_thread->is_JfrSampler_thread(), "should be JFRSampler");`
would be a little more clean.